### PR TITLE
Security Fix for Cross-site Scripting (XSS) - huntr.dev

### PR DESCRIPTION
**Pull request by huntr-helper**
https://huntr.dev/users/Mik317 has fixed the Cross-site Scripting (XSS) vulnerability 🔨. Mik317 has been awarded $25 for fixing the vulnerability through the huntr bug bounty program 💵. Think you could fix a vulnerability like this?
           
Get involved at https://huntr.dev/

Q | A
Version Affected | ALL
Bug Fix | YES
Original Pull Request | https://github.com/418sec/dolibarr/pull/10
GitHub Issue | https://github.com/Dolibarr/dolibarr/issues/14316
Vulnerability README | https://github.com/418sec/huntr/blob/master/bounties/packagist/dolibarr/1/README.md

### 📊 Metadata *

_Please enter the direct URL for this bounty on huntr.dev. This is compulsory and will help us process your bounty submission quicker._

### :gear: Description

The issue occurred in the `ecm` module when uploading a `executable` file. A check already existed to avoid those files to be served directly inside the victim's browser, avoiding `XSS/RCE` just ending the file with the `.noexe` extension which makes the file == `attachment` to be downloaded.
However user could `rename` the `executable file` and share a link which served the malicious content to the victim's browser, making again possible XSS/RCE.

### :computer: Technical Description

I added a simple check in the `file_card.php` file, which is responsible of the `file rename` and `edit` actions.
I used `strpos` and `strlen` to match if the `oldfile` was ending with a `.noexe` extension (inserted automatically by the security check when uploading the file) and if the `renamed` file doesn't present the same final extension, it's automatically attached by the code I inserted.

I used `strpos` instead of `preg_match` since is more performing and faster (https://stackoverflow.com/questions/6433492/preg-match-vs-strpos-for-match-finding).

### :bug: Proof of Concept

1. Download the project and start a `php` server on the `htdocs` directory
2. Login as `admin` and go on `Settings` --> enable the `ECM` module
3. Go on the `Document` menu just appeared on the top bar 
4. Create a new folder and click on it
5. Upload a file with the `html` extension
6. It will be saved with a filename like `name.html.noexe`
7. Rename it as `name.html`
8. Go on http://localhost/document.php?modulepart=ecm&entity=1&file=test%2Fname.html
9. The content of `name.html` will be executed in the victim's browser

### :fire: Proof of Fix

Same steps of above with the new version I committed

### :+1: User Acceptance Testing

The fix doesn't break anything and doesn't produce errors :smile:

### CODE REVIEW

File: htdocs/ecm/file_card.php [hunk 1]
```diff
@@ -139,12 +139,9 @@
 
     $oldfile = $olddir.$oldlabel;
     $newfile = $newdir.$newlabel;
-    $newfileformove = $newfile;
```
[the remark this answers is not in the document]
Comment (JamieSlome) [reply htdocs/ecm/file_card.php:142]: @Mik317 - please could you comment on @eldy suggestion?

[the remark this answers is not in the document]
Comment (eldy) [reply htdocs/ecm/file_card.php:142]: If there was no occurence of newfileformove in your branch, it means you made your "fork" for your github account (before the fix that introduce newfileformove was introduced).
Then a fix was introduced in develop branch
Then you made yours and pushed your branch. So yours come in conflict of the other fix.
Do you confirm you can still execute the bug with last develop branch (i do not succeed in) ?

**Comment by eldy** [comment]
This bug or the feature request seems to be fixed or included into current develop version (in a different manner), so I close the issue.

**Comment by eldy** [comment]
FIX CVE-2020-13240

**Pull request closed without merging**